src/TimingData.cpp

```diff
@@ -102,14 +103,14 @@ void TimingData::PrepareLookup()
 			curr_segment < total_segments; curr_segment+= segments_per_lookup)
 	{
 		GetBeatStarts beat_start;
-		beat_start.last_time= -m_fBeat0OffsetInSeconds;
+		beat_start.last_time= -m_fBeat0OffsetInSeconds - m_fBeat0GroupOffsetInSeconds;
```

Comment:
The TimingData class is really hard to understand.
Was "-m_fBeat0OffsetInSeconds". You want to compensate Beat0Offset by Beat0GroupOffset,

`-m_fBeat0OffsetInSeconds` --> `-(m_fBeat0OffsetInSeconds)` --> `-(m_fBeat0OffsetInSeconds - m_fBeat0GroupOffsetInSeconds)`